src/agentscope/service/multi_modality/dash_scope.py

```diff
@@ -0,0 +1,229 @@
+import requests
+
+
+import dashscope
+from dashscope.audio.tts import SpeechSynthesizer
+
+
+from agentscope.service.service_response import ServiceResponse
+from agentscope.service.service_status import ServiceExecStatus
+
+
+def dashscope_text_to_image(
+    prompt: str,
+    api_key: str,
+    number_of_images: int = 1,
+    size: str = "1024*1024",
+    model: str = "wanx-v1",
+    saved_dir: str = "./figs",
+) -> ServiceResponse:
+    """Generate an image based on a text prompt.
+
+    Args:
+        prompt (`str`): the text prompt.
+        api_key (`str`): The api key for the dashscope api.
+        number_of_images (`int`, defaults to `1`): the number of images
+        to generate.
+        size (`str`, defaults to `1024*1024`): size of the image
+        model (`str`, defaults to 'wanx-v1'): the model to use.
+        saved_dir (`str`, defaults to './figs'): the directory to save the
+        generated images.
+    Returns:
+        ServiceResponse:
+        A dictionary with two variables: `status` and`content`.
+        If `status` is ServiceExecStatus.SUCCESS,
+        the `content` is a dict with key 'fig_paths" and
+        value is a list of the paths to the generated images.
+    Example:
+        prompt = "A beautiful sunset in the mountains"
+        print( dashscope_text_to_image(prompt)) gives:
+        {'status': 'SUCCESS',
+        'content': {'fig_paths':
+            ['FIG_RELATEIVE_PATH1',
+            'FIG_RELATEIVE_PATH2']}}
+    """
+
+    dashscope.api_key = api_key
+    res = dashscope.ImageSynthesis.call(
+        model=model,
+        prompt=prompt,
+        n=number_of_images,
+        size=size,
+    )
+    if res.status_code == HTTPStatus.OK:
+        if res.output.task_status == "SUCCEEDED":
+            results = res.output.results
+            # write iamge to disk
+            # if saved_dir does not exist, create it
+            fig_paths = []
+            if not os.path.exists(saved_dir):
+                os.makedirs(saved_dir)
+            for i, result in enumerate(results):
+                fig_path = f"{saved_dir}/{prompt}_{i}.png"
+                with open(fig_path, "wb+") as f:
+                    f.write(requests.get(result.url).content)
+                fig_paths.append(fig_path)
+            return ServiceResponse(
+                ServiceExecStatus.SUCCESS,
+                {"file_paths": fig_paths},
+            )
+        else:
+            err_msg = f"Task failed with status {res.output.task_status}"
+            err_msg += res.message
+            return ServiceResponse(
+                ServiceExecStatus.ERROR,
+                {"error": err_msg},
+            )
+    else:
+        err_msg = f"Error in calling the API: {res.status_code}"
+        err_msg += res.message
+        return ServiceResponse(
+            ServiceExecStatus.ERROR,
+            {"error": err_msg},
+        )
+
+
+def dashscope_image_to_text(
+    image_urls: Union[str, Tuple[str, ...]],
+    prompt: str,
```

Comment:
The docstring misses the parameter `prompt`, and should we consider to provided a default prompt here? e.g. describe the figure.